runtime/src/consensus/tendermint/verifier/mod.rs

```diff
@@ -339,48 +336,18 @@ impl Verifier {
         runtime_header: Header,
         epoch: EpochTime,
     ) -> Result<ConsensusState, Error> {
-        // Verify runtime ID matches.
-        if runtime_header.namespace != self.runtime_id {
-            return Err(Error::VerificationFailed(anyhow!(
-                "header namespace does not match trusted runtime id"
-            )));
-        }
-
-        if runtime_header.round < cache.last_verified_round {
-            // Ignore requests for earlier rounds.
-            return Err(Error::VerificationFailed(anyhow!(
-                "round seems to have moved backwards"
-            )));
-        }
-        if epoch < cache.last_verified_epoch {
-            // Ignore requests for earlier epochs.
-            return Err(Error::VerificationFailed(anyhow!(
-                "epoch seems to have moved backwards"
-            )));
-        }
-
-        // If round has advanced make sure that consensus height has also advanced as a round can
-        // only be finalized in a subsequent consensus block. This is to avoid a situation where
-        // one would keep feeding the same consensus block for subsequent rounds.
-        //
-        // NOTE: This needs to happen before the call to verify_consensus_only which updates the
-        //       cache.last_verified_height field.
-        if runtime_header.round > cache.last_verified_round
-            && consensus_block.height <= cache.last_verified_height
-        {
-            return Err(Error::VerificationFailed(anyhow!(
-                "consensus height did not advance but runtime round did"
-            )));
-        }
+        // Perform basic verifications.
+        predicates::verify_namespace(self.runtime_id, &runtime_header)?;
+        predicates::verify_round_advance(&cache, &runtime_header, &consensus_block, epoch)?;
+        predicates::verify_consensus_advance(&cache, &consensus_block)?;
 
         // Verify the consensus layer block first to obtain an authoritative state root.
-        let consensus_block = self.verify_consensus_only(cache, instance, consensus_block)?;
-        let state_root = consensus_block.get_state_root();
-        let state = ConsensusState::from_protocol(
-            self.protocol.clone(),
-            state_root.version + 1,
-            state_root,
-        );
+        let height = consensus_block.height;
```

Comment:
Can we trust this height? Before these changes the height was taken from the trusted header. Maybe validation should be added to `verify_consensus_only`.
```
// Validate height.
if height != consensus_block.height {
     return Err(Error::VerificationFailed(anyhow!("height mismatch")));
}
```

[nit] No need to introduce a new variable.

Reply:
Updated.